Build gitian using Ubuntu 18.04 based docker containers

sickpig opened the pull request:
WIP check symbol is failing: 

```
Checking glibc back compat...
test/test_bitcoin: symbol logf from unsupported version GLIBC_2.27
qt/bitcoin-qt: symbol logf from unsupported version GLIBC_2.27
Makefile:11138: recipe for target 'check-symbols' failed
make: Leaving directory '/home/ubuntu/build/bitcoin/distsrc-x86_64-linux-gnu/src'
make: *** [check-symbols] Error 1
```

ken2812221 committed:
Change gitian-descriptors to use bionic instead (core PR 13171)

sickpig committed:
Enable C++14 for Berkeley DB, boost and protobuf

sickpig committed:
Doc to produce BU gitian binaries via docker based on Ubuntu 18.04

sickpig committed:
Further GCC-7 and glibc-2.27 compat code tweaks

sickpig commented:
@gandrewstone taking out of WIP, PR is ready to go in imho.

gandrewstone commented on doc/gitian-building-bu.md:
```diff
@@ -0,0 +1,30 @@
+Build the binaries
+------------------
+
+cd ~/src/gitian-builder
```
needs ```bash

sickpig replied:
right, fixing.

sickpig replied:
fixed

sickpig committed:
Fix doc formatting

gandrewstone commented on doc/gitian-building-bu.md:
````diff
@@ -0,0 +1,34 @@
+```bash
+sudo apt install git apt-cacher-ng ruby docker.io
+sudo usermod -a -G docker $USER
+cd ~/src
````
```gnome-session-quit```  # log out and back in so group change takes effect

sickpig replied:
Actually issuing this command:

```bash
exec su -l $USER
```
from the same terminal that will be used to run the `bin/gbuild` should do trick.

sickpig replied:
fixed

sickpig committed:
Add a way to make usermod command effective without reopening a session

gandrewstone commented on contrib/gitian-descriptors/gitian-win.yml:
```diff
@@ -2,13 +2,13 @@
 name: "BUcash-win-1.5"
 enable_cache: true
 suites:
-- "trusty"
```
bionic

sickpig replied:
fixed

sickpig committed:
Fix win gitian descriptor so that it uses the correct distro

gandrewstone commented on depends/packages/bdb.mk:
```diff
@@ -9,7 +9,7 @@ define $(package)_set_vars
 $(package)_config_opts_mingw32=--enable-mingw
 $(package)_config_opts_linux=--with-pic
-$(package)_cxxflags=-std=c++11
+$(package)_cxxflags=-std=c++1y
```
c++14, 1y is the pre-standard flag

sickpig replied:
you are right indeed.

sickpig replied:
fixed

gandrewstone commented on depends/packages/boost.mk:
```diff
@@ -20,7 +20,7 @@ $(package)_archiver_$(host_os)=$($(package)_ar)
 $(package)_archiver_darwin=$($(package)_libtool)
 $(package)_config_libraries=chrono,filesystem,program_options,system,thread,test
-$(package)_cxxflags=-std=c++11 -fvisibility=hidden
+$(package)_cxxflags=-std=c++1y -fvisibility=hidden
```
c++14

sickpig replied:
ditto

sickpig replied:
fixed

gandrewstone commented on depends/packages/protobuf.mk:
```diff
@@ -4,7 +4,7 @@ $(package)_download_path=$(native_$(package)_download_path)
 $(package)_sha256_hash=$(native_$(package)_sha256_hash)
 $(package)_dependencies=native_$(package)
-$(package)_cxxflags=-std=c++11
+$(package)_cxxflags=-std=c++1y
```
14

sickpig replied:
ditto

sickpig replied:
fixed

sickpig committed:
Use the proper label for the C++14 standard

C++1y is a name that was used to refer to the C++14
standard before it was completed. now that C++14 is
final there's no need to use 1y.

sickpig committed:
We need to use posix variant of mingw to get win binaries built

sickpig committed:
Apply fix_qt571_win32_qmake.patch for static library builds
